Source/DafnyCore/Backends/GoLang/GoCodeGenerator.cs

```diff
@@ -179,9 +205,22 @@ private struct Import {
       return w;
     }
 
-    protected override void DependOnModule(string moduleName, bool isDefault, ModuleDefinition externModule,
+    protected override void DependOnModule(Program program, ModuleDefinition module, ModuleDefinition externModule,
       string libraryName) {
-      var import = CreateImport(moduleName, isDefault, externModule, libraryName);
+      var goModuleName = "";
+      if (GoModuleMode && program.Compilation.AlreadyCompiledRoots.Contains(module.Tok.Uri)) {
+        var dooManifest = DooFile.Read(module.Tok.Uri.AbsolutePath).Manifest;
+        dooManifest.Options.TryGetValue(Options.Backend.TargetId, out var targetOptions);
+        object moduleName = null;
+        if (targetOptions is TomlTable tomlTable) {
+          tomlTable.TryGetValue(CommonOptionBag.BackendModuleName.Name, out moduleName);
+        }
+        goModuleName = moduleName is string name ? moduleName + "/" : "";
+        if (String.IsNullOrEmpty(goModuleName)) {
+          Options.ErrorWriter.WriteLine($"Go Module Name not found for the module {module.GetCompileName(Options)}");
```

Comment:
Use `Error(...)` instead so that you get a clean compilation error and eventually produce a non-zero exit code.

Reply:
I've kept this because it might make sense to generate non module code than not generate anything. Lmk if you think otherwise.